### DISCUSSION TITLE
Cob sk6812 5 wire leds wiring help

### DISCUSSION THREAD

**Discussion by TechsavvyCNC** [awawa-dev/HyperHDR#975] [Q&A]
I purchased fcob leds but they have five wires.  Does anybody know how to wire that to hyperhdr?  BTF-LIGHTING FCOB COB LED Strip RGBW 6500K Flexible High Density Uniform Light Kit Bluetooth APP Music IR Remote Controller 16.4FT 12mm Width Color Changing for Bedroom Kitchen Home Indoor Decoration https://a.co/d/4vldPw3

**Comment by awawa-dev** [comment]
Hi
These LEDs are not SK6812 LEDs or addressable LEDs.

**Reply by TechsavvyCNC** [reply]
so there no  way for these to work?

**Reply by awawa-dev** [reply]
Not with HyperHDR, they can only have one color set across the entire length.

**Comment by TechsavvyCNC** [comment]
I pretty sure it does multiple colors across the length.  Maybe the control box has something to do with it.  Is there anyway to try and connect five wires to hyperhdr or hyperspi?  Or do only 3 wire leds work?

**Reply by awawa-dev** [reply]
This is not a rule: SPI and ws2813 have 4 wires/lanes, other Neopixel usually 3 wires. I don't know what LEDs you have, it looks like some analog PWM LED type. But maybe I'm wrong. Usually users choose sk6812 RGBW 5V which is 3 wires.

**Reply by TechsavvyCNC** [reply]
Mine are 12v.  How do you wire 4 wire leds?  I've only seen 3 wire diagrams.

**Reply by TechsavvyCNC** [reply]
Dang.  Your right it only does 1 color at a time.  I still wanna know how to hook up 4 wires leds.  I'll have to get these then.  https://www.amazon.com/dp/B0DCVMVCRN/ref=sspa_dk_detail_0?pd_rd_i=B0DCVMVCRN&pd_rd_w=3UEcP&content-id=amzn1.sym.f2f1cf8f-cab4-44dc-82ba-0ca811fb90cc&pf_rd_p=f2f1cf8f-cab4-44dc-82ba-0ca811fb90cc&pf_rd_r=K630CNEED1KQHT7H6DPZ&pd_rd_wg=Mapno&pd_rd_r=6a973931-f3ba-4eb2-9a1e-1ee694c07b54&s=hi&sp_csd=d2lkZ2V0TmFtZT1zcF9kZXRhaWxfdGhlbWF0aWM&th=1

**Reply by TechsavvyCNC** [reply]
Or would these be better?  BTF-LIGHTING FCOB WS2814 IC RGBW 6000K COB LED Strip UL Listed Flexible High Density Uniform Chasing Color Light 16.4FT DC24V 784LED/m 14IC/m 10mm Width for Home Indoor Decor(No Adapter or Controller) https://a.co/d/0B89Ztv

**Reply by awawa-dev** [reply]
I haven't used such LEDs, note another trap: `Group of 56LED addressable. 1 pixel=1 IC chip=56 LED` . There is WLED Discord for examples of using many different LEDs, here users tend to choose sk6812.

**Reply by TechsavvyCNC** [reply]
So these would work?  BTF-LIGHTING RGBW RGBCW White SK6812 (Similar WS2812B) 16.4ft 5m 60leds/pixels/m Individually Addressable Flexible 4 color in 1 LED Dream Color LED Strip Non-waterproof DC5V https://a.co/d/7X71hvX

**Reply by awawa-dev** [reply]
version  60leds/meter, Rgb+cool White, IP30, 5v looks fine for TV application

**Reply by TechsavvyCNC** [reply]
You want cool white right?

**Reply by awawa-dev** [reply]
yes

**Comment by TechsavvyCNC** [comment]
I bought the correct leds.  I'm using a rpi 2040 is this how it's suppose to hook up
![20241106_040203](https://github.com/user-attachments/assets/c9eb7826-08c7-438e-9fac-3e9fe47e909d)

**Comment by TechsavvyCNC** [comment]
And the white wire from the pi connects to the ground  on the rp2040 right?

**Comment by awawa-dev** [comment]
@TechsavvyCNC 
I'm sorry but I can't figure out what's what in the picture and how it's all connected. Or what firmware you're using but since you soldered some cable to pin 14 you probably have to use custom firmware.

**Reply by TechsavvyCNC** [reply]
![17309456849306530341510044001362](https://github.com/user-attachments/assets/234daa7b-6737-4b92-a320-37a00d412e35)

**Comment by TechsavvyCNC** [comment]
14 is the led utput.  2 is for clock 3 is data and 5 is spi chip select.  I'm using a rp2040 zero.

**Comment by TechsavvyCNC** [comment]
[
![17309457551703398450970557211971](https://github.com/user-attachments/assets/c6cf8a1d-2b94-4ee5-a3f8-1e11cc3d02b0)
![17309458084904837542868858110218](https://github.com/user-attachments/assets/c8d44bb9-21c0-42c2-b7d1-c4efe60dc929)
](url)

**Comment by TechsavvyCNC** [comment]
Should i have it connected where clock is 26, data is 28 and spi select is 29.  Led control is gpio14/gpio16.

**Comment by awawa-dev** [comment]
> 14 is the output. 2 is for clock 3 is data and 5 is spi chip select. I'm using a rp2040 zero.

Maybe it's typo but data is GPIO 4 not 3. First check if you have a communication with Pico then start connecting LEDs: https://github.com/awawa-dev/HyperSPI?tab=readme-ov-file#performancedebug-output

**Comment by awawa-dev** [comment]
Connecting ground this way on Pico you are risking a short-circuit with neighboring 3.3V or 5V (worst case scenario). Isolate it properly.

**Comment by TechsavvyCNC** [comment]
Where should pico ground be isolated?

**Comment by TechsavvyCNC** [comment]
just let me get this right.  if using hrdpico i just plug usb from pi to pi zero and connect those 4 wires in your pictures.  if using hyperspi then i have to externally power the pi zero and connect those 4 wires.  your picture shows the white wire going to ground.
![image](https://github.com/user-attachments/assets/2d36ad96-cb6d-4349-9f9d-8c13edec3668)
i think i had connector backwards

**Reply by AstaRom** [reply]
> if using hyperspi then i have to externally power the pi zero and connect those 4 wires. your picture shows the white wire going to ground.

You can power the rp2040 board directly from the power supply.
Connection example:
- [Visual scheme](https://github.com/AstaRom/AstaRom.github.io/blob/main/files/board/ambilight.png)
- [Printed circuit board](https://github.com/awawa-dev/HyperSPI/discussions/54#discussioncomment-10304776) (simple option)
- [Printed circuit board](https://github.com/AstaRom/AstaRom.github.io/blob/main/files/img/pages/rp2040_2.jpg) (more complex version)

**Comment by TechsavvyCNC** [comment]
Im externally power the rp2040 so i can remove the white wire?  Is hyperspipico better?_
